src/Knp/Component/Pager/Event/Subscriber/Paginate/Doctrine/ORM/QueryBuilderSubscriber.php

```diff
@@ -12,7 +12,14 @@ public function items(ItemsEvent $event)
     {
         if ($event->target instanceof QueryBuilder) {
             // change target into query
-            $event->target = $event->target->getQuery();
+            $queryBuilder = $event->target;
+
+            // Remove "order by" part for the count query for performance purpose
+            $countQueryBuilder = clone $queryBuilder;
```

Comment:
shouldn't the RDBM apply optimisations  on the query? maybe an analysis of the query plan would help understand why it's slow?